api.ts

```diff
@@ -1699,11 +1739,41 @@ export interface SyncTableOptions<
    * sync tables that have a dynamic schema.
    */
   dynamicOptions?: DynamicOptions;
+
   /**
-   * A function to autocomplete values for properties marked with `mutable` and `autocomplete`.
+   * An autocomplete function to use for any dynamic schema properties.
+   * The name of the property that's being modified by the doc editor
+   * is available in autocomplete function's context parameter.
+   *
+   * @example
+   * ```
+   * coda.makeDynamicSyncTable({
+   *   name: "MySyncTable",
+   *   getSchema: async function (context) => {
+   *     return coda.makeObjectSchema({
+   *       properties: {
+   *         dynamicPropertyName: {
+   *           type: coda.ValueType.String,
+   *           mutable: true,
+   *           autocomplete: coda.AutocompleteValueType.Dynamic,
+   *         },
+   *       },
+   *     });
+   *   },
+   *   autocomplete: async function (context) => {
+   *     if (context.propertyName === "dynamicPropertyName") {
+   *       return ["Dynamic Value 1", "Dynamic value 2"];
+   *     }
+   *     throw new coda.UserVisibleError(
+   *       `Cannot autocomplete property ${context.propertyName}`
+   *     );
+   *   },
+   *   ...
+   * ```
+   *
    * @hidden
    */
-  propertyAutocomplete?: PropertyAutocompleteMetadataFunction;
+  autocomplete?: PropertyAutocompleteMetadataFunction<any>;
```

Comment:
Should this go in `dynamicOptions`, since that's the set of all things that apply only to dynamic schemas?

Reply:
I originally had it in dynamicOptions, but then realized we support dynamic properties in static sync tables.

For example, the Jira issues table actually uses `makeSyncTable` rather than `makeDynamicSyncTable`. Seems like dynamic sync tables always want you to give them a URL link before they finish being created and static sync tables don't?

Reply:
Right, but when you wan to declare a dynamic schema on a static table, you still use `dynamicOptions`. Here's the Jira def: https://github.com/coda/packs/blob/5d18547bb9c13965cdd1114978150f32f2e3aa7c/packs/atlassian/jira/common/formulas.ts#L522

Reply:
Ah, makes sense. Moved `autocomplete` into `dynamicOptions` for non-dynamic sync tables. For a dynamic sync table it remains top-level